reports/report_generator.py

```diff
@@ -0,0 +1,39 @@
+def bar_charts_for_temperature(readings_data_structure, date):
+    """generates a bar chart of maximum and minimum temperature on each day of a month
+
+    Args:
+        data ([Dictionary]): [Nested dictionary of all data]
+    """
+    date = date.split('/')
+    year = date[0]
+    month = date[1]
+    if int(month) < 10:
+        month = month[1]
```

Comment:
Should printing of bar charts have any dependency on date? They should be getting values of temperature and that's it. This is the violation of SRP.